qiskit/opflow/gradients/natural_gradient.py

```diff
@@ -120,43 +113,71 @@ def convert(
         # Instantiate the QFI metric which is used to re-scale the gradient
         metric = self._qfi_method.convert(operator[-1], params) * 0.25
 
-        # Define the function which compute the natural gradient from the gradient and the QFI.
         def combo_fn(x):
-            c = np.real(x[0])
-            a = np.real(x[1])
-            if self.regularization:
-                # If a regularization method is chosen then use a regularized solver to
-                # construct the natural gradient.
-                nat_grad = NaturalGradient._regularized_sle_solver(
-                    a, c, regularization=self.regularization
-                )
-            else:
-                try:
-                    # Try to solve the system of linear equations Ax = C.
-                    nat_grad = np.linalg.solve(a, c)
-                except np.linalg.LinAlgError:  # singular matrix
-                    nat_grad = np.linalg.lstsq(a, c)[0]
-            return np.real(nat_grad)
+            return self.nat_grad_combo_fn(x, self.regularization)
 
         # Define the ListOp which combines the gradient and the QFI according to the combination
         # function defined above.
         return ListOp([grad, metric], combo_fn=combo_fn)
 
+    @staticmethod
+    def nat_grad_combo_fn(x, regularization=None):
+        """
+        Natural Gradient Function Implementation
+        Args:
+            x: Iterable consisting of Gradient, Quantum Geometric Tensor
+            regularization: Regularization method
+
+        Returns: Natural Gradient
+
+        """
+        c = x[0]
+        a = x[1]
+        if any(np.abs(np.imag(c_item)) > 1e-8 for c_item in c):
+            raise Warning("The imaginary part of the gradient are non-negligible.")
```

Comment:
Is this supposed to end the program or just print a warning? If the latter, then we should use `warnings.warn` as `raise Warning` actually stops the script 🙂 If it should actually stop the program we should use an appropriate error, like for instance `ValueError`.